src/structures/MessageComponentInteraction.js

```diff
@@ -51,6 +51,18 @@ class MessageComponentInteraction extends Interaction {
     this.webhook = new InteractionWebhook(this.client, this.applicationID, this.token);
   }
 
+  /**
+   * The component which was interacted with
+   * @type {?MessageActionRowComponent|Object}
+   * @readonly
+   */
+  get component() {
+    return this.message.components
+      .map(row => row.components)
+      .flat()
```

Comment:
Is `flat` defaulted to Infinity?

Reply:
Default depth is 1, though not sure it makes a difference here based on the data the API gives us.